monitor.py

```diff
@@ -529,20 +533,21 @@ def get_failed_monitor_testcases(spy_link,job_type):
                 response_2 = requests.get(test_log_url,verify=False, timeout=15)
                 if response_2.status_code == 200:
                     data = response_2.json()
-                    e2e_failure_list = data['Tests']
-                    return e2e_failure_list
+                    for tc in data['Tests']:
+                        monitor_tc_failures.append(tc["Test"]["Name"])
                 else:
-                    return "Failed to get response from e2e-test log file url!"
+                    error_msg = "Failed to get response from e2e-test log file url!"
             else:
-                return "Test summary file not found"
+                error_msg = "Test summary file not found"
         else:
-            return "Failed to get response from e2e-test directory url"
+            error_msg = "Failed to get response from e2e-test directory url"
     except requests.Timeout:
-        return "Request timed out"
+        error_msg = "Request timed out"
     except requests.RequestException:
-        return "Error while sending request to url"
+        error_msg = "Error while sending request to url"
```

Comment:
Instead of introducing a new variable error_msg, you could change all return statements to return two params as shown in suggestion right ?
```suggestion
        return monitor_tc_failures, "Error while sending request to url"
```